src/bin/server.rs

```diff
@@ -0,0 +1,7 @@
+use saito_rust::server;
+use tokio::signal;
+
+#[tokio::main]
+pub async fn main() -> saito_rust::Result<()> {
+    server::run(signal::ctrl_c()).await
+}
```

Comment:
Should we just put this in main.rs??

I definitely don't like src/bin/, I'd expect to find binaries in there....

Are we going to have more than one entry point? I think you can just replace main.rs with this...

Reply:
the `src/bin` setup allows us flexibility in kicking off the system. In the future if we wanted to implement a cli, we could implement a `src/bin/cli.rs` to help facilitate that.

Perhaps this is getting a bit ahead of ourselves however, so for now we'll just settle for a classic `main.rs`